src/common/buffer.cc

```diff
@@ -1118,6 +1118,12 @@ static simple_spinlock_t buffer_debug_lock = SIMPLE_SPINLOCK_INITIALIZER;
     return true;
   }
 
+  bool buffer::list::can_zero_copy(const char *dst) const
```

Comment:
how about is_provided_buffer? I'd like to avoid the ambiguity of two can_zero_copy()s with different meanings